python/core/auto_generated/qgsscalecalculator.sip.in

```diff
@@ -65,6 +65,16 @@ Calculate the scale denominator
 :param canvasWidth: Width of the map canvas in pixel (physical) units
 
 :return: scale denominator of current map view, e.g. 1000.0 for a 1:1000 map.
+%End
+
+    double calculateCanvasWidth( const QgsRectangle &mapExtent, double scale ) const;
```

Comment:
I think we could do better with units and terms.
1. we are mixing rectangle/size with double/length
2. this works for with or height, so the term `width` shouldn't be encoded into the naming

Option a:

```suggestion
    QSizeF calculateCanvasSize( const QgsRectangle &mapExtent, double scale ) const;
```

Option b:

```suggestion
    // Usually pass in `mapExtent.width()` for your requirements
    double canvasLengthToPixelLength( double canvasLength, double scale ) const;
```

Reply:
I love option A.

Reply:
Can we replace "canvas" with "image"?

Reply:
Done.